11-Backbonejs/Views-of-Collections.md

```diff
@@ -0,0 +1,208 @@
+# Views of Collections
+
+We have created a view for an individual task.  However views can also be built to manage a collection.  Therefore our next step is to create a TaskListView.  This view will manage everything we have on the site to including the form to add a new task to the list.  If you examine the image below the TaskListView's area of concern is indicated in yellow.  
+
+![TaskListView](images/Ada-TaskList-JS.png)
+
+## Learning Goals
+-  Create a view to manage the entire TaskList collection
+-  Create event listeners to add new models to the collection
+-  Create event listeners for custom Backbone Events
+-  Create a `render` method which creates and renders a group of smaller views.  
+
+## Getting Started
+
+This view starts like `TaskView` with a file named `/src/views/task_list_view.js`.  We can move our renderList code into the view's render method and make a few adjustments.
+
+```javascript
+import Backbone from 'backbone';
+import _ from 'underscore';
+import $ from 'jquery';
+import TaskView from '../views/task_view.js';
+
+
+var TaskListView = Backbone.View.extend({
+  initialize: function(params) {
+    this.template = params.template;
+  },
+  render: function() {
+    // Clear the unordered list
+    this.$('.todo-items').empty();
+    // Iterate through the list rendering each Task
+    var that = this;
+    this.model.each(function(task) {
+      // Create a new TaskView with the model & template
+      var taskView = new TaskView({
+        model: task,
+        template: that.template,
+        tagName: 'li'
+      });
+      // Then render the TaskView
+      // And append the resulting HTML to the DOM.
+      that.$('.todo-items').append(taskView.render().$el);
+    });
+    return this;
+  }
+});
+
+export default TaskListView;
+```
+Just like the `TaskView` in `initialize` we store the template in an attribute of the view.  The view's render method looks like the render method we wrote earlier with a few differences particularly with regard to `this` and `that`.
+
+### A Note about `this` and `that`.  
+
+If you'll notice we save our current context `this` into a variable named `that` in the render method.  We do this because the `.each` method's callback method is called within the context of the collection.  
```

Comment:
I've seen it both ways, and originally learned `that = this;` in the dark ages before learning Ruby.  The Backbone book I read used `that` and some of the examples I've seen in Stackoverflow use `self`.  So...